lib/pbench/test/unit/server/test_unpack_tarballs.py

```diff
@@ -0,0 +1,117 @@
+import glob
+from pathlib import Path
+import pytest
+
+from pbench.common.logger import get_pbench_logger
+from pbench.server.filetree import FileTree, TarballUnpackError
+from pbench.server.unpack_tarballs import Unpack
+
+
+@pytest.fixture()
+def make_logger(server_config):
+    """
+    Construct a Pbench Logger object
+    """
+    return get_pbench_logger("TEST", server_config)
+
+
+class TestUnpackTarballs:
+    tar_name = "A.tar.xz"
+    tarlist = [tar_name]
+    linksrc = "TO-UNPACK"
+
+    def mock_glob(self, path):
+        """Returns the Tarball list"""
+        return TestUnpackTarballs.tarlist
+
+    def mock_resolve(self: Path, strict: bool = False):
+        """
+        Return a "full" fake file name
+
+        Args:
+            strict: Not used in the mock
+
+        Returns:
+            new Path with fake resolution
+        """
+        return Path(self.tar_name)
```

Comment:
Given that these mocks are trival (e.g., they aren't checking their inputs), I would use lambda expressions instead.  That way, the reader can see what the mock is going to do without having to page up to the function definitions.